dronecan-gui-tool: init at 1.2.28-unstable-2025-06-17

xyven1 opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

xyven1 committed:
python3Packages.easywebdav: init at 1.2.0

xyven1 committed:
python3Packages.install-freedesktop: init at 0.1.2

xyven1 committed:
python3Packages.pymonocypher: init at 4.0.2.2

xyven1 committed:
python3Packages.qtwidgets: init at 1.2

xyven1 committed:
dronecan-gui-tool: init at 1.2.28-unstable-2025-06-17

xzfc commented on pkgs/by-name/dr/dronecan-gui-tool/package.nix:
```diff
@@ -0,0 +1,71 @@
+    convert icons/logo.ico icons/dronecan_gui_tool.png
+    mv $out/share/applications/DroneCAN\ GUI\ Tool.desktop $out/share/applications/dronecan_gui_tool.desktop
+    substituteInPlace $out/share/applications/dronecan_gui_tool.desktop \
+      --replace "Exec=/dronecan_gui_tool-1.2.25.data/scripts/dronecan_gui_tool" "Exec=dronecan_gui_tool"
```
Don't include the version `1.2.25` verbatim. Maybe use `{version}` or replace it using regex.

xzfc commented on pkgs/by-name/dr/dronecan-gui-tool/package.nix:
```diff
@@ -0,0 +1,71 @@
+  ];
+
+  dependencies = with python3Packages; [
+    pkgs.iproute2
```
Put it outside of `with python3Packages;`.

xyven1 commented:
Bump @xzfc

FliegendeWurst commented on pkgs/development/python-modules/pymonocypher/default.nix:
```diff
@@ -0,0 +1,24 @@
+    setuptools-scm
+  ];
+
+  dependencies = [numpy];
```
missing meta

FliegendeWurst commented on pkgs/development/python-modules/install-freedesktop/default.nix:
```diff
@@ -0,0 +1,21 @@
+    setuptools-scm
+  ];
+
+  doCheck = false;
```
missing meta

FliegendeWurst commented on pkgs/development/python-modules/easywebdav/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+  dependencies = [requests];
+
+  doCheck = false;
```
missing meta

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+  dependencies = [pyqt5];
+
+  doCheck = false;
```
missing meta

xyven1 commented:
@FliegendeWurst Any particular reason you added the comments (specifically are you able to finally get this merged), or were you just scanning for basic issues like missing `meta` sections across a ton of PRs?

FliegendeWurst commented:
> or were you just scanning for basic issues like missing `meta` sections across a ton of PRs?

I am going through some older PRs, trying to find out why they were not merged. In this case the missing meta attributes were one obvious reason. I don't have permission to merge myself, but usually someone who can comes along relatively quickly after I review.

FliegendeWurst commented on pkgs/development/python-modules/easywebdav/default.nix:
```diff
@@ -0,0 +1,33 @@
+  meta = {
+    description = "A straight-forward WebDAV client, implemented using Requests";
+    homepage = "https://github.com/amnong/easywebdav";
+    license = with lib.licenses; [ isc ];
```
```suggestion
    license = with lib.licenses; [ isc ];
    maintainers = with lib.maintainers; [ xyven1 ];
```
New packages must have a maintainer.

FliegendeWurst commented on pkgs/development/python-modules/easywebdav/default.nix:
```diff
@@ -0,0 +1,33 @@
+  pythonImportsCheck = [ "easywebdav" ];
+
+  meta = {
+    description = "A straight-forward WebDAV client, implemented using Requests";
```
```suggestion
    description = "Straight-forward WebDAV client, implemented using Requests";
```

FliegendeWurst commented on pkgs/development/python-modules/install-freedesktop/default.nix:
```diff
@@ -0,0 +1,28 @@
+  meta = {
+    description = "Setuptools extension to install freedesktop.org app icons";
+    homepage = "https://github.com/welshjf/install_freedesktop";
+    license = with lib.licenses; [ asl20 ];
```
```suggestion
    license = with lib.licenses; [ asl20 ];
    maintainers = with lib.maintainers; [ xyven1 ];
```

FliegendeWurst commented on pkgs/development/python-modules/pymonocypher/default.nix:
```diff
@@ -0,0 +1,34 @@
+    license = with lib.licenses; [
+      cc0
+      bsd2
+    ];
```
```suggestion
    ];
    maintainers = with lib.maintainers; [ xyven1 ];
```

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,31 @@
+  meta = {
+    description = "Custom widget library for PyQt5 and PySide2 (Qt for Python). Free to use in your own applications.";
+    homepage = "https://github.com/pythonguis/python-qtwidgets";
+    license = with lib.licenses; [ mit ];
```
```suggestion
    license = with lib.licenses; [ mit ];
    maintainers = with lib.maintainers; [ xyven1 ];
```

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,31 @@
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "xyven1";
```
does not match meta.homepage

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,31 @@
+  doCheck = false;
+
+  meta = {
+    description = "Custom widget library for PyQt5 and PySide2 (Qt for Python). Free to use in your own applications.";
```
```suggestion
    description = "Custom widget library for PyQt5 and PySide2";
```

FliegendeWurst commented on pkgs/development/python-modules/easywebdav/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+  dependencies = [ requests ];
+
+  doCheck = false;
```
```suggestion
  doCheck = false; # no tests
```

FliegendeWurst commented on pkgs/development/python-modules/install-freedesktop/default.nix:
```diff
@@ -0,0 +1,28 @@
+    setuptools-scm
+  ];
+
+  doCheck = false;
```
```suggestion
  doCheck = false; # no tests

  pythonImportsCheck = [ "install_freedesktop" ];
```

FliegendeWurst commented on pkgs/development/python-modules/pymonocypher/default.nix:
```diff
@@ -0,0 +1,34 @@
+  ];
+
+  dependencies = [ numpy ];
+
```
```suggestion

  doCheck = true;
  nativeCheckInputs = [ pytestCheckHook ];
```

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  dependencies = [ pyqt5 ];
+
+  doCheck = false;
```
```suggestion
  doCheck = false; # no tests
```

xyven1 commented:
@FliegendeWurst Ah, I see. 

The detailed comments are very much appreciated! I think I addressed all the concerns, so this PR is a good a shape as it has ever been.

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,37 @@
+buildPythonPackage {
+  pname = "qtwidgets";
+  version = "1.2";
+  format = "setuptools";
```
```suggestion
  pyproject = true;
```

xyven1 replied:
The use of `format="setuptools"';` was very specific, with your change you get the following error:
```
ERROR Backend 'setuptools.build_meta:__legacy__' is not available.
```

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,37 @@
+}:
+buildPythonPackage {
+  pname = "qtwidgets";
+  version = "1.2";
```
```suggestion
  version = "1.2-unstable-YYYY-MM-DD";
```
The version is from a commit so you need to put the date here.

xyven1 replied:
Done.

FliegendeWurst commented on pkgs/development/python-modules/qtwidgets/default.nix:
```diff
@@ -0,0 +1,37 @@
+    ./fix-missing-svgs.patch
+  ];
+
+  build-inputs = [ setuptools-scm ];
```
```suggestion
  # Remove install_requires on markdown. It is not used.
  postPatch = ''
    substituteInPlace setup.py --replace-fail "'markdown'," ""
  '';

  build-system = [ setuptools-scm ];
```
Or amend the second patch

xyven1 replied:
Whats the reasoning for this?

FliegendeWurst replied:
`pyproject = true` is the preferred build method for Python packages. In this case, the package states a dependency on markdown that makes no sense (it is not used anywhere).
So remove the dependency declaration.
